src/contracts/core/Slasher.sol

```diff
@@ -1,102 +1,368 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.12;
 
-import "../interfaces/ISlasher.sol";
-import "../interfaces/IDelegationManager.sol";
-import "../interfaces/IStrategyManager.sol";
-import "../libraries/StructuredLinkedList.sol";
-import "../permissions/Pausable.sol";
-import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
-import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "./SlasherStorage.sol";
 
+import "../libraries/EpochUtils.sol";
+import "../libraries/SlashingAccountingUtils.sol";
 
 /**
- * @notice This contract is not in use as of the Eigenlayer M2 release.
- *
- * Although many contracts reference it as an immutable variable, they do not
- * interact with it and it is effectively dead code. The Slasher was originally
- * deployed during Eigenlayer M1, but remained paused and unused for the duration
- * of that release as well.
- *
- * Eventually, slashing design will be finalized and the Slasher will be finished
- * and more fully incorporated into the core contracts. For now, you can ignore this
- * file. If you really want to see what the deployed M1 version looks like, check
- * out the `init-mainnet-deployment` branch under "releases".
- *
- * This contract is a stub that maintains its original interface for use in testing
- * and deploy scripts. Otherwise, it does nothing.
- */ 
-contract Slasher is Initializable, OwnableUpgradeable, ISlasher, Pausable {
-     
-    constructor(IStrategyManager, IDelegationManager) {}
+ * @notice Slasher
+ */
+contract Slasher is SlasherStorage {
+    constructor(
+        IStrategyManager _strategyManager,
+        IDelegationManager _delegationManager,
+        IOperatorSetManager _operatorSetManager
+    ) SlasherStorage(_strategyManager, _delegationManager, _operatorSetManager) {}
 
-    function initialize(
-        address,
-        IPauserRegistry,
-        uint256
-    ) external {}
+    /**
+     * @notice Called by an AVS to increase its own slashing request for a given
+     * operator set and operator in the current epoch
+     *
+     * @param operator the operator that the calling AVS is to increase the bips they want to slash
+     * @param operatorSetID the id of the operator set the AVS is increasing their slashing for
+     * @param strategies the list of strategies slashing requested is being modified for
+     * @param bipsToIncrease the basis points slashing to modify for given strategies
+     */
+    function increaseRequestedBipsToSlash(
```

Comment:
Not convinced we need an "increase" method. IMO we should have:
* `requestSlash`
* `reduceSlash`
* `executeSlash`